### DISCUSSION TITLE
brew commands fail to build gem native extension

### DISCUSSION THREAD

**Discussion by BastianZim** [Homebrew/discussions#923] [Getting started] [answered]
Hi,

I tried to run `brew upgrade` but received the error linked in this gist: https://gist.github.com/BastianZim/89af23cc93c77738ec3982f17b208875

I also ran `brew doctor`, `brew man`, `brew --version` and `brew config` but received the above error as well. 

The environment is a MacBook Pro running 11.2.1.

Any idea why this is happening?

Thanks!

**Comment by whyaname** [comment]
Getting the same error, hopefully get a resolution soon!

**Reply by BastianZim** [reply]
Happened to me only after updating to, I think, 3.0.2. Is that the same for you or has that happened on previous versions as well?

**Reply by whyaname** [reply]
Only with the latest update yes, same as you. Also running a MacBook Pro macOS 11.2.1

**Comment by Bo98** [comment, marked as answer]
Can I get a bit of information on your setup?

What do these return:

* `xcrun --sdk macosx --show-sdk-path`
* `xcode-select -p`
* `ls -l /Library/Developer/CommandLineTools/SDKs`
* `ls -l /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs`

(it's ok if some error - just let me know what they say)

**Reply by whyaname** [reply]
Hi, I am getting the same error (exactly the same), so I hope its okay if I paste this info, here is the gist:

https://gist.github.com/whyaname/991991b3b7377069cd8b36a28e48f254

**Reply by carlocab** [reply]
Are you running macOS 10.15?

**Reply by BastianZim** [reply]
Running `xcrun --sdk macosx --show-sdk-path` gave me: `xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun`.

I fixed that by running `xcode-select --install` and now everything works, thanks @Bo98!

Bit weird though, because I had the command line tools installed already and everything worked before `3.0.2`.  Could it be that the update uninstalled something?

**Reply by whyaname** [reply]
> Are you running macOS 10.15?

Nope, 11.2.1

**Reply by Bo98** [reply]
Homebrew doesn't uninstall anything, but a macOS update might have (I'm aware it does on ARM at least).

**Reply by BastianZim** [reply]
> > Are you running macOS 10.15?
> 
> Nope, 11.2.1

Have you tried running `xcode-select --install` again? Somehow fixed it for me.

**Reply by whyaname** [reply]
> > > Are you running macOS 10.15?
> > 
> > 
> > Nope, 11.2.1
> 
> Have you tried running `xcode-select --install` again? Somehow fixed it for me.

Says command line tools are already installed, quite confusing haha

**Reply by BastianZim** [reply]
> Homebrew doesn't uninstall anything, but a macOS update might have (I'm aware it does on ARM at least).

Hmm, strange, I'm on intel and didn't update macOS in-between. Anyways, fixed now. 😄

**Reply by BastianZim** [reply]
> > > > Are you running macOS 10.15?
> > > 
> > > 
> > > Nope, 11.2.1
> > 
> > 
> > Have you tried running `xcode-select --install` again? Somehow fixed it for me.
> 
> Says command line tools are already installed, quite confusing haha

Hmm, mine were too but it still fixed it. 😄

Did you try running `xcrun --sdk macosx --show-sdk-path`?

**Reply by whyaname** [reply]
> > > > > Are you running macOS 10.15?
> > > > 
> > > > 
> > > > Nope, 11.2.1
> > > 
> > > 
> > > Have you tried running `xcode-select --install` again? Somehow fixed it for me.
> > 
> > 
> > Says command line tools are already installed, quite confusing haha
> 
> Hmm, mine were too but it still fixed it. 😄
> 
> Did you try running `xcrun --sdk macosx --show-sdk-path`?

Yep, shows me the path to the SDK, that's different from the output you got.

**Reply by Bo98** [reply]
@whyaname 

Try reinstalling the CLT:

```
sudo rm -rf /Library/Developer/CommandLineTools
xcode-select --install
```

**Reply by carlocab** [reply]
macOS updates get rid of CLT installations now, at least on Big Sur, so that's why your CLT installation went missing, @BastianZim.

@whyaname, you have the wrong CLT installed. It's for macOS 10.15 but you should have the one for Big Sur. Run
```
sudo rm -rf /Library/Developer/CommandLineTools
xcode-select --install
```

**Edit:** 32 seconds late. Oops.

**Reply by whyaname** [reply]
I see, thank you so much @Bo98 @carlocab

**Reply by BastianZim** [reply]
Well, that's nice of Apple... Thanks for the info though, @carlocab!
